src/ResourceManager/Relay/Commands.Relay/Cmdlets/AuthorizationRule/NewAzureRelayKey.cs

```diff
@@ -0,0 +1,104 @@
+        [Parameter(ParameterSetName = WcfRelayAuthoRuleParameterSet)]
+        [Parameter(ParameterSetName = HybridConnectionAuthoRuleParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string Namespace { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2, ParameterSetName = WcfRelayAuthoRuleParameterSet,
+            HelpMessage = "WcfRelay Name.")]
+        [ValidateNotNullOrEmpty]
+        public string WcfRelay { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2, ParameterSetName = HybridConnectionAuthoRuleParameterSet,
+            HelpMessage = "HybridConnection Name.")]
+        [ValidateNotNullOrEmpty]
+        public string HybridConnection { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 3,
+            HelpMessage = "AuthorizationRule Name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true,
+            HelpMessage = "Regenerate Keys - 'PrimaryKey'/'SecondaryKey'.")]
+        [ValidateSet(RegeneKeys.PrimaryKey,
+            RegeneKeys.SecondaryKey,
+            IgnoreCase = true)]      
+        public string RegenerateKey { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var regenKey = new RegenerateKeysParameters(RegenerateKey);
+
+
+            // Generate new Namespace List Keys for the specified AuthorizationRule
+            if (ParameterSetName == NamespaceAuthoRuleParameterSet)
+            {
+                if (ShouldProcess(target: RegenerateKey, action: string.Format(Resources.RegenerateKeyNamesapce, Name, Namespace)))
+                {
+                    WriteObject(Client.NamespaceRegenerateKeys(ResourceGroupName, Namespace, Name, RegenerateKey));
+                }
+            }
+
+            // Generate new WcfRelay List Keys for the specified AuthorizationRule
+            if (ParameterSetName == WcfRelayAuthoRuleParameterSet)
+            {
+                if (ShouldProcess(target: RegenerateKey, action: string.Format(Resources.RegenerateKeyWcfRelay, Name, WcfRelay)))
+                {
+                    WriteObject(Client.WcfRelayRegenerateKeys(ResourceGroupName, Namespace, WcfRelay, Name, RegenerateKey));
+                }
+            }
+
+            // Generate new WcfRelayHybridConnection List Keys for the specified AuthorizationRule
+            if (ParameterSetName == HybridConnectionAuthoRuleParameterSet)
+            {
+                if (ShouldProcess(target: RegenerateKey, action: string.Format(Resources.RegenerateKeyHybirdconnection, Name, HybridConnection)))
+                {
+                    WriteObject(Client.HybridConnectionsRegenerateKeys(ResourceGroupName, Namespace, HybridConnection, Name, RegenerateKey));
```

Comment:
Should enumerate the keys if this is a collection (for each of these branches), bu using ```WriteObject(collection, true)```